### PR TITLE
Basic gtkdoc support

### PR DESCRIPTION
**Pull request by jpakkane**
Create and install documentation with gtkdoc. This is a very simple implementation and we might want to get comments from actual users before merging.

Docs are not compiled on plain ninja command. Doing so would be wasteful because most of the time you don't care about docs during development. You can build them manually with 'ninja modulename-doc'.

Documentation is always regenerated on install. Wasting some cycles is acceptable here to ensure that no stale documentation files are installed by accident.

Also change the core so that there can be more than one custom install script.

### CODE REVIEW

File: gtkdochelper.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# Copyright 2012-2014 The Meson development team
```
Comment (ignatenkobrain) [inline gtkdochelper.py]: I think 2015

**Comment by ignatenkobrain** [comment]
test fails. 
`FileNotFoundError: [Errno 2] No such file or directory: 'gtkdoc-scan'`

I think it should be added via `find_program()`, not in compile stage

**Comment by jpakkane** [comment]
Is gtk-doc installed on the CI server?

**Comment by jpakkane** [comment]
retest this please

**Comment by jpakkane** [comment]
Is this ok to land now? It works at least enough for appstream-glib.

**Pull request merged**